File: examples/PRB_93_155403_2016/main.cpp
#include "setup.h"

int main(){
  // initialize simulation
  Ptr<SimulationGrating> s = SimulationGrating::instanceNew();
  s->setPeriodicity(1e-6);
  s->setGx(50);
  // initialize materials
  s->addMaterial("Au", "fullGold.txt");
  s->addMaterial("Vacuum", "fullVacuum.txt");
  // initialize layers
  s->addLayer("BottomAir", 0, "Vacuum");
  s->addLayer("GoldSubstrateBottom", 0.5e-6, "Au");
  s->addLayer("GoldGratingBottom", 5e-6,"Au");
  s->setLayerPatternGrating("GoldGratingBottom", "Vacuum", 0.5e-6, 0.2e-6);

  s->addLayer("VacGap", 1e-6, "Vacuum");
  s->addLayerCopy("GoldGratingTop", "GoldGratingBottom");
  s->addLayerCopy("GoldSubstrateTop", "GoldSubstrateBottom");
  s->addLayerCopy("TopAir", "BottomAir");
  // set source and probe
  s->setSourceLayer("GoldGratingBottom");
  s->setSourceLayer("GoldSubstrateBottom");

  s->setProbeLayer("VacGap");

  // set output

  s->setKxIntegralSym(500);
  s->setKyIntegralSym(200, 5);
  //s->useInverseRule();
  s->getSysInfo();
  s->optPrintIntermediate();
  s->setThread(4);
  s->build();
  s->run();
  s->saveToFile("gold_to_vac.txt");
  // std::cout << s->getPhiAtKxKy(0, 0, 0) << std::endl;
  return 0;
}

File: examples/PRB_93_155403_2016/main.lua
-- this is a lua realization of the corresponding cpp file
s = SimulationGrating.new()
s:SetLattice(1e-6);
s:AddMaterial("Au", "fullGold.txt");
s:AddMaterial("Vacuum", "fullVacuum.txt");

s:AddLayer("BottomAir", 0, "Vacuum");
s:AddLayer("GoldSubstrateBottom", 0.5e-6, "Au");
s:AddLayer("GoldGratingBottom", 5e-6,"Au");
s:SetLayerPatternGrating("GoldGratingBottom", "Vacuum", 0.5e-6, 0.2e-6);

s:AddLayer("VacGap", 1e-6, "Vacuum");
s:AddLayerCopy("GoldGratingTop", "GoldGratingBottom");
s:AddLayerCopy("GoldSubstrateTop", "GoldSubstrateBottom");
s:AddLayerCopy("TopAir", "BottomAir");

s:SetSourceLayer("GoldSubstrateBottom");
s:SetSourceLayer("GoldGratingBottom");
s:SetProbeLayer("VacGap");
s:OutputSysInfo();

s:OptPrintIntermediate("test");
s:SetThread(4);
s:SetNumOfG(101);
s:SetKxIntegralSym(500);
s:SetKyIntegralSym(200, 5);
s:InitSimulation();
s:IntegrateKxKy();
--print(string.format("%e", s:GetPhiAtKxKy(0, 0, 0)));
